service/k8sclient/internal/kubectl/proxy.go

```diff
@@ -0,0 +1,137 @@
+// dbaas-controller
+// Copyright (C) 2020 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Package kubectl provides kubectl CLI wrapper.
+package kubectl
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/rand"
+	"net"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/avast/retry-go"
+	"github.com/percona/pmm/utils/pdeathsig"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
+var reservedProxyPorts = new(sync.Map)
+
+// reserveProxyPort reserves a ramdom proxy port from range [10000, 19999).
+// It stores the proxy command under the reserved port so we can get back to it later.
+func (k *KubeCtl) reserveProxyPort(cmd *exec.Cmd) string {
+	random := rand.New(rand.NewSource(time.Now().UnixNano()))
+	var port int
+	for {
+		port = 10000 + random.Intn(10000)
+		_, loaded := reservedProxyPorts.LoadOrStore(strconv.Itoa(port), cmd)
+		if !loaded {
+			break
+		}
+	}
+	return strconv.Itoa(port)
+}
+
+// RunProxy runs kubectl proxy on port that is returned.
+func (k *KubeCtl) RunProxy(ctx context.Context) (string, error) {
+	var port string
+	// Try to run kubectl proxy on random port.
+	err := retry.Do(
```

Comment:
`retry` uses `context.Background()` as the default context. 
Should we specify the context like in the [documentation example](https://github.com/avast/retry-go#func--context)?

Reply:
Good point, thanks. I added the context option.